utils/docker/Dockerfile.checkpatch

```diff
@@ -0,0 +1,39 @@
+# Copyright (C) 2021 Intel Corporation
+# All rights reserved.
+#
+# 'recipe' for Docker to build a image for running checkpatch.
+#
+
+# Pull base image
+FROM centos:centos7
```

Comment:
If you are going to have a dedicated Dockerfile for this, why not use a distro with *much* newer tools, like, say, `fedora:latest`?

Reply:
I tried quite hard with this PR to use centos8, for that very reason, but I couldn't find a way of getting ShellCheck installed, it seems to have moved from yum/dnf to using snap as an installer which is new to me and doesn't seem to work in docker.  Doing a periodic update to latest would benefit here, but having tried I took the safe option and extracted the code/versions we're already using.

Reply:
I am using Shellcheck and even some of the other linters that Daos does not currently have enabled with a Fedora:latest Dockerfile for other projects.

I have not tried the Checkpatch yet.

Reply:
> I tried quite hard with this PR to use centos8,

centos8 is going to eventually have the same geriatric issues that centos7 has currently.  The whole "EL" project simply is not updated frequently enough with newer (not just bug-fixed) software as it ages and therefore suitable for this task.

> for that very reason, but I couldn't find a way of getting ShellCheck installed

It will be installed with `fedora:latest`.  I use Fedora as my main platform and I use ShellCheck all the time on it.  It does catch way more than El7's ageing ShellCheck, FWIW.

> it seems to have moved from yum/dnf to using snap as an installer

That wouldn't be the "official" EL8 ShellCheck.  The Snap store version is likely not maintained by RH and in this manner is a "third-party" store.  That said, I am quite pessimistic on both Snap and Flatpak packaging systems, as they are both _bundling_ packaging systems, which the core software you want to install also installs copies of all of it's dependencies.  That then puts a burden and responsibility on those package maintainers to not only update their packages when the core software is updated, but they also have to be paying attention to the upstreams of **all** of the software they are bundling and put out new releases when any of those upstreams release a security update.  I am **not at all** convinced that those packagers are taking on the responsibility sufficiently and actually paying attention to their bundled software upstreams.

https://bugzilla.redhat.com/show_bug.cgi?id=1772274 would be the RH perspective on ShellCheck for EL8.

> which is new to me and doesn't seem to work in docker.

And we don't want to use, anyway, IMO, for the reasons above.

> Doing a periodic update to latest would benefit here, but having tried I took the safe option and extracted the code/versions we're already using.

I would say that if we are going to this effort to have a dedicated linting Dockerfile, let's do it with recent tool versions, and we are only going to be able to get recent tool versions with something like Fedora.

Reply:
> > I tried quite hard with this PR to use centos8,
> 
> centos8 is going to eventually have the same geriatric issues that centos7 has currently. The whole "EL" project simply is not updated frequently enough with newer (not just bug-fixed) software as it ages and therefore suitable for this task.
> 
> > for that very reason, but I couldn't find a way of getting ShellCheck installed
> 
> It will be installed with `fedora:latest`. I use Fedora as my main platform and I use ShellCheck all the time on it. It does catch way more than El7's ageing ShellCheck, FWIW.
> 
> > it seems to have moved from yum/dnf to using snap as an installer
> 
> That wouldn't be the "official" EL8 ShellCheck. The Snap store version is likely not maintained by RH and in this manner is a "third-party" store. That said, I am quite pessimistic on both Snap and Flatpak packaging systems, as they are both _bundling_ packaging systems, which the core software you want to install also installs copies of all of it's dependencies. That then puts a burden and responsibility on those package maintainers to not only update their packages when the core software is updated, but they also have to be paying attention to the upstreams of **all** of the software they are bundling and put out new releases when any of those upstreams release a security update. I am **not at all** convinced that those packagers are taking on the responsibility sufficiently and actually paying attention to their bundled software upstreams.
> 
> https://bugzilla.redhat.com/show_bug.cgi?id=1772274 would be the RH perspective on ShellCheck for EL8.
> 
> > which is new to me and doesn't seem to work in docker.
> 
> And we don't want to use, anyway, IMO, for the reasons above.

Agreed, on all counts.  I didn't find that redhat ticket but it does explain why I had problems.

> > Doing a periodic update to latest would benefit here, but having tried I took the safe option and extracted the code/versions we're already using.
> 
> I would say that if we are going to this effort to have a dedicated linting Dockerfile, let's do it with recent tool versions, and we are only going to be able to get recent tool versions with something like Fedora.

That wasn't the problem I set out to solve, but it seems splitting this dockerfile out is a essential first step for us to get there.